modules/oci-build/image_tool/append_layers.go

```diff
@@ -0,0 +1,155 @@
+package main
+
+import (
+	"archive/tar"
+	"bytes"
+	"io"
+	"io/fs"
+	"log/slog"
+	"os"
+	"path/filepath"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/layout"
+	"github.com/google/go-containerregistry/pkg/v1/match"
+	"github.com/google/go-containerregistry/pkg/v1/mutate"
+	"github.com/google/go-containerregistry/pkg/v1/tarball"
+	"github.com/spf13/cobra"
+)
+
+var CommandAppendLayers = cobra.Command{
+	Use:   "append-layers oci-path [path-to-tarball...]",
+	Short: "Appends a tarball or directory to every image in an OCI index.",
+	Args:  cobra.MinimumNArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		oci := args[0]
+		extra := args[1:]
+
+		if len(extra) == 0 {
+			return
+		}
+
+		layers := []v1.Layer{}
+		for _, path := range extra {
+			layers = append(layers, loadLayerFromDirOrTarball(path))
+		}
+
+		appendLayersToAllImages(oci, layers...)
+	},
+}
+
+func loadLayerFromDirOrTarball(path string) v1.Layer {
+	stat, err := os.Stat(path)
+	must("could not open directory or tarball", err)
+
+	var layer v1.Layer
+	if stat.IsDir() {
+		var buf bytes.Buffer
+
+		tw := tar.NewWriter(&buf)
+
+		filepath.Walk(path, func(target string, info fs.FileInfo, err error) error {
+			must("walk error", err)
+
+			header, err := tar.FileInfoHeader(info, info.Name())
```

Comment:
@ThatsMrTalbot should we maybe overwrite the file owner and file ModTime, AccessTime, ... here?
So this becomes a reproducible build?
We also have the SOURCE_DATE_EPOCH environment variable that is used with ko.

Reply:
I think we should also write some tests for this tool (maybe not as part of this PR).